### PR TITLE
EW - Remove Restaurants Backend and Frontend

### LINKED ISSUE #18: BACKEND: Remove Restaurant Backend and Frontend

### PR DESCRIPTION
**Pull request by eug-wong**
**Description:**
- This PR removes Restaurants from the Navbar, as well as all relevant frontend files for creating, editing, and indexing Restaurants.
- Modifies Navbar test cases to test Placeholder instead

**Tests:**
- Passes all tests, including mutation tests

Closes #18

### CODE REVIEW

**Comment by divyagsubramonian** [comment]
LGTM! All the files tied to Restaurants were deleted and changes to corresponding files (NavBar) were made accordingly.

**Review by divyagsubramonian — APPROVED** [review]
LGTM! All the files tied to Restaurants were deleted and changes to corresponding files (NavBar) were made accordingly.

**Comment by Ramon-W** [comment]
Manually approved pending UI Tests.

**Review by pconrad — APPROVED** [review]
LGTM

**Comment by pconrad** [comment]
To fix the testing gaps in this PR, see: 

https://ucsb-cs156-f24.slack.com/archives/C080JHJN9QE/p1732427414430999

**Comment by pconrad** [comment]
When you remove both the Restaurant *and* the UCSBDates backend, it creates test coverage gaps that need to be fixed.

I took care of the merge conflicts for you (which I probably shouldn't have done—but I was being nice).  But as a result, the problem is now exposed.

Here's a solution that works.

https://ucsb-cs156-f24.slack.com/archives/C080JHJN9QE/p1732427414430999

**Review by kanav9063 — APPROVED** [review]
correctly made changes based on professors advice to remove Restaurants Backend and Frontend

File: src/main/java/edu/ucsb/cs156/rec/controllers/DummyController.java [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+public class DummyController extends ApiController {
+
+    @GetMapping("")
+    public String getById(@RequestParam Long id) throws EntityNotFoundException {
+        if (id == 1) {
+            return "String1";
+        }
+        throw new EntityNotFoundException(String.class, id);
+    }
+}
```
Comment (pconrad) [inline src/main/java/edu/ucsb/cs156/rec/controllers/DummyController.java:26]: This file should be placed under `src/test/java/edu/ucsb/cs156/rec/controllers/DummyController.java` not `src/main/java/edu/ucsb/cs156/rec/controllers/DummyController.java`

**Review by pconrad — CHANGES_REQUESTED** [review]
One small change needed: `DummyController.java` goes under `test` , not `main`

**Review by divyagsubramonian — DISMISSED** [review]
Looks good! DummyController was moved to the correct directory.

**Review by pconrad — DISMISSED** [review]
LGTM

**Pull request merged**